src/idl_gen_js.cpp

```diff
@@ -524,10 +523,30 @@ class JsGenerator : public BaseGenerator {
     return "NS" + std::to_string(HashFnv1a<uint64_t>(file.c_str()));
   }
 
-  static std::string GenPrefixedImport(const std::string &full_file_name,
-                                       const std::string &base_file_name) {
+  std::string GenPrefixedImport(const std::string &full_file_name,
+                                const std::string &base_name) {
+    // either keep the include path as it was
+    // or use only the base_name + kGeneratedFileNamePostfix
+    std::string path;
+    if (parser_.opts.keep_include_path) {
+      auto it = parser_.included_files_.find(full_file_name);
+      FLATBUFFERS_ASSERT(it != parser_.included_files_.end());
+      path =
+          flatbuffers::StripExtension(it->second) + kGeneratedFileNamePostfix;
+    } else {
+      path = base_name + kGeneratedFileNamePostfix;
+    }
+
+    // add the include prefix
+    path = flatbuffers::ConCatPathFileName(parser_.opts.include_prefix, path);
+
+    if (!flatbuffers::IsAbsPath(path)) {
```

Comment:
again, this doesn't look like it tests anything meaningful on Windows.

Reply:
The if here is to test if its an absolute path so we do not prefix with "./" as that was the old behavior.
Only relative paths should be prefixed, in my opinion... (thats why this AbsPath, test)

Reply:
I dont know for sure! :) But we can skip this by just leaving it as it was and only support relative paths =) any thing else as absolut paths is senseless